src/core/simplify.rkt

```diff
@@ -35,90 +50,57 @@
   (->* (expr? #:rules (listof rule?))
        (#:precompute boolean? #:prune boolean?)
        expr?)
+  (if use-egg-math?
+      (simplify-batch-egg exprs #:rules rls #:precompute precompute?)
+      (begin
+        (warn 'simplify #:url "faq.html#egg-herbie"
+              "Falling back on racket egraph because egg-herbie package not installed")
+        (simplify-batch-herbie-egraph exprs #:rules rls #:precompute precompute?))))
+
+(define/contract (simplify-batch-egg exprs #:rules rls #:precompute precompute?)
+  (-> (listof expr?) #:rules (listof rule?) #:precompute boolean? (listof expr?))
   (debug #:from 'simplify (format "Simplifying:\n  ~a" (string-join (map ~a exprs) "\n  ")))
 
-  (define start-time (current-inexact-milliseconds))
-  (define eg (mk-egraph))
-  (define ens (for/list ([expr exprs]) (mk-enode-rec! eg expr)))
-  (define ex (apply mk-extractor ens))
-
-  (define phases
-    (filter identity
-            (list (rule-phase rls)
-                  (and precompute? precompute-phase)
-                  (and prune? prune-phase))))
+  (local-require "eggmath.rkt")
 
+  (define start-time (current-inexact-milliseconds))
+  (define res
```

Comment:
We don't need the define block here. And I don'e believe we use `start-time-inner` below.